### PR TITLE
Support download history

### PR DESCRIPTION
**Pull request by NomadThanatos**
1. Support download history, it can batch download all books which are not in history, and save them to history automatically.
2. Change filename from "#{book_id}_#{title}.epub" to "#{title}.epub"

### CODE REVIEW

File: lib/readmoo_dl/lister.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      @api = API.new(username: username, password: password)
+      @books = {}
+    end
+
+    def list()
+        result = [];
+        JSON.parse(@api.list())['included'].each do |item|
+            result.push(item.slice("id", "title"))
+        end
+        result
```
Comment (msglight4874) [inline lib/readmoo_dl/lister.rb:13]:
```suggestion
        JSON.parse(@api.list())['included'].map do |item|
            item.slice('id', 'title')
        end
```

File: main.rb.example [hunk 2]
```diff
@@ -15,3 +15,25 @@ downloader = ReadmooDL::Downloader.new(username, password)
...
+# Remove the corresponding line in history.txt if you want to re-download a specific one.
+=begin
+lister = ReadmooDL::Lister.new(username, password)
+history = {}
+File.open('history.txt', 'r') do |f|
+    while line = f.gets
+        history_item = JSON.parse(line)
+        history[history_item['id']] = history_item['title']
+    end
+end
```
Comment (msglight4874) [inline main.rb.example:29]:
```suggestion
File.readlines('history.txt').each do |line|
  history_item = JSON.parse(line)
  history[history_item['id']] = history_item['title']
end
```

File: lib/readmoo_dl/lister.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module ReadmooDL
+  class Lister
+    def initialize(username, password)
+      @api = API.new(username: username, password: password)
+      @books = {}
+    end
+
+    def list()
```
Comment (msglight4874) [inline lib/readmoo_dl/lister.rb:8]:
```suggestion
    def list
```

File: main.rb.example [hunk 4]
```diff
@@ -15,3 +15,25 @@ downloader = ReadmooDL::Downloader.new(username, password)
...
+
+lister.list().each do |item|
+    if history[item['id']].nil?
+        downloader.download(item['id'])
+        File.open('history.txt', 'a') do |f|
+            f.puts item.to_json
+        end
+    end
+end
```
Comment (msglight4874) [inline main.rb.example:36]:
```suggestion
File.open('history.txt', 'a') do |history_file|
  lister.list.each do |item|
    next unless history[item['id']]
    downloader.download(item['id'])
    history_file.puts(item.to_json)
  end
end
```

Comment (NomadThanatos) [reply main.rb.example]: In my tests, I think it should be
```next unless history[item['id']].nil?```

**Pull request merged**